model/post.go

```diff
@@ -736,10 +736,18 @@ func (o *Post) ToNilIfInvalid() *Post {
 	return o
 }
 
-func (o *Post) ForPlugin() *Post {
-	p := o.Clone()
-	p.Metadata = nil
-	return p
+func (o *Post) RemovePreviewPost() {
+	if o.Metadata == nil || o.Metadata.Embeds == nil {
+		return
+	}
+	n := 0
+	for _, embed := range o.Metadata.Embeds {
+		if embed.Type != PostEmbedPermalink {
+			o.Metadata.Embeds[n] = embed
+			n++
+		}
+	}
+	o.Metadata.Embeds = o.Metadata.Embeds[:n]
```

Comment:
Instead of using `n` to keep track of the position manually, you can just set `newEmbeds := o.Metadata.Embeds[:0]` and use `append` to add the elements.

Reply:
This is a revert PR :) Let's do those in the actual PR along with the fix.

Reply:
Yeah, missed that. Sorry.

Reply:
@noxer I am ignoring this comment in https://github.com/mattermost/mattermost-server/pull/20686 (the revert of the revert) since `RemovePreviewPost` is no longer used, the entire metadata is removed.